runtime/src/builtins.rs

```diff
@@ -163,6 +181,8 @@ fn builtin_feature_transitions() -> Vec<BuiltinFeatureTransition> {
                 "zk_token_proof_program",
                 solana_zk_token_sdk::zk_token_proof_program::id(),
                 solana_zk_token_proof_program::process_instruction,
+                125906, // the average CUs for 6 zkp program instructions,
+                        // see comment at https://github.com/solana-labs/solana/pull/30639
```

Comment:
Sorry to come back to this, but I'd prefer setting an extremely high default so that the zk program is forced to always manually consume compute units, maybe even `u64::MAX`. It would stink to open up a DOS opportunity just because we forget to consume compute units in a new instruction. With `u64::MAX`, it should fail tests immediately, right?

@samkim-crypto what do you think?

Reply:
Yep, if the idea is to use the default compute budget when the individual instructions fail to manually consume compute units, then making it always fail by setting it to a high number would be the way to go. 

Currently (well, it is currently behind the feature gate), the proof program does [consume](https://github.com/solana-labs/solana/blob/master/programs/zk-token-proof/src/lib.rs#L37) units at the program level, so it is not completely clear when the default units will kick in. I was hoping to see how this is done in the follow-up PR.

Reply:
This is a good concern. Think about it, the risk of new instruction (of an existing program) would use much more units than pre-define program default cost applies not just zkp program, but to all. 

It is always preferred builtin programs to manually deduct units during process. But for those don't have the luxury to do so at the moment, a default projects the worst scenario should be good enough. 

For zkp program, my reason for using average instead of worse scenario (407_121 CU) is that compute budget defines default unit per instruction as 200_000, and max as 1_400_000 cu per tx. So I think anything close to 200_000 is reasonable, cause you can't have more than 7 of that in single tx. But if we really want to be careful, then worse scenario is ok too. u64::max might be overkill - that'd fail tx immediately. This is my 2 cents.

Reply:
@samkim-crypto looks we just crossed each other :)

The use case of default cost is described in this comment https://github.com/solana-labs/solana/pull/30639#discussion_r1129740453. Basically if builtin.process_instruction() does not deduct units from compute budget, then the program's default cost will be used.

Reply:
> if the idea is to use the default compute budget when the individual instructions fail to manually consume compute units

regarding this, currently no builtin manually consumes units (i mean nothing was deducted from compute budget after builtin.process_instruction()). So if sets default cost to max, then all tx has builtin will fail.

Reply:
(BTW, I think this discussion is very helpful, thanks guys)

Reply:
Ok that all makes sense. How about we split the difference, and use the CUs from `VerifyTransfer` as the default? That should be the most used instruction. And since it's over 200k, we'll have to request more compute units in tests, which means we're much less likely to forget.

Reply:
sounds good to me, changed it in comment 959f686